### PR TITLE
Ephemeral Environments for PRs on MindsDB

### PR DESCRIPTION
**Pull request by ShrutiC-git**
## Description

This PR is adding support for preview environments for PR events (opened, sync'ed, reopened, etc) on MindsDB, powered by [Uffizzi](https://www.uffizzi.com/).

**Fixes** https://github.com/mindsdb/mindsdb/issues/4150

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [X] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📄 This change requires a documentation update

### What is the solution?

This PR is using GHA workflows to build and provision the preview environments. Utilizing the docker-compose example from MindsDB, the PR uses docker-compose.uffizzi.yml to build MindsDB, the DB-handler, and the ml-handler from source - this makes sure the latest changes are reflected in the preview. The docker-compose also includes a seeded Postgres DB (seeded with Iris dataset, against which the PoC was tested) and Nginx-ingress.

Once a PR event is triggered — PR opened, PR synced, review requested — the build workflow will build MindsDB and related handlers, and the preview workflow will either spin up a new preview or update an existing preview.  The previews are ephemeral — so they live for only as long as needed.

A comment will be posted on the PR containing the URL of the preview environment, allowing contributors and maintainers to easily navigate to the preview.

To test this, the first step was to seed the Postgres with Iris-dataset. The `seed.sql` script manages that. Post that, when I opened a PR against my fork of Minds, the `uffizzi-build.yml` file triggered the `Build PR Image` action. Once this action was completed, the `uffizzi-preview.yml` GHA-file triggered the `Deploy Preview` action. _We use this 2-step workflow for open-source projects, to allow open-source projects to have previews for contributions from forks. [Here](https://www.uffizzi.com/preview-environments-guide/github-actions-two-stage-workflow) is a more in-depth read into why we use a two-stage workflow._

- [Here is the PR](https://github.com/ShrutiC-git/mindsdb/pull/3) opened against my personal fork of Minds
- [Here is the comment](https://github.com/ShrutiC-git/mindsdb/pull/3#issuecomment-1358050586) that was posted on the PR
- [Here is the preview environment](https://app.uffizzi.com/github.com/ShrutiC-git/mindsdb/pull/3)

The preview environment will take you to the preview. To test if I was able to connect to my `Postgres DB, named iris`, please run the following script:

> CREATE DATABASE uffizzi_example
> WITH ENGINE = "postgres",
> PARAMETERS = {
>     "user": "postgres",
>     "password": "postgres",
>     "host": "localhost",
>     "port": "5432",
>     "database": "iris"
>     };

This will connect you to the Postgres instance, that comes with `docker-compose.uffizzi.yml`. Once connected to the DB, I was able to build a predictor model, look at the Data-Insights, and run predictions on the preview environment. 
PS: I also had a lot of fun playing around with MindsDB.

If there is anything that needs to be added/changed, or any other feedback, please do let me know! Looking forward to powering preview environments for Minds.

## Checklist:

- [X] My code follows the style guidelines(PEP 8) of MindsDB.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation, or created issues to update them.
- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] I have checked that my code additions will fail neither code linting checks nor unit test.
- [X] I have shared a short loom video or screenshots demonstrating any new functionality.

### CODE REVIEW

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb_native/blob/stable/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   <sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

**Comment by ShrutiC-git** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ShrutiC-git** [comment]
hey there, @ZoranPandovski, hope you were able to go over the PR ~ what are your thoughts, do let me know!

**Comment by ShrutiC-git** [comment]
@ZoranPandovski, friendly ping - could I please get an estimate on when you'll be able to take a look into this?

If this is not something the project is interested in, let me know and I'll close the PR.

**Comment by ZoranPandovski** [comment]
Sorry for the late response @ShrutiC-git, we can always try this for sure. Please just merge staging so it resolves the confilcts

**Comment by ShrutiC-git** [comment]
copy, @ZoranPandovski. 

Will clean this out. Thanks

**Comment by ShrutiC-git** [comment]
Fixed the conflicts, @ZoranPandovski. I had kept some files which were deleted in the `staging`; building the handlers without them were throwing errors. We can keep iterating on this if this is something the community is interested in. Let me know

File: requirements.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ waitress >= 1.4.4
 pymongo[srv] >= 4.1.1
 setuptools
 psutil
-sqlalchemy >= 1.4.29, < 2.0.0
+sqlalchemy >= 1.1.0, < 1.5
```
Comment (StpMax) [inline requirements.txt:14]: why did you need to change the sqlalchemy version?

**Review by StpMax — CHANGES_REQUESTED** [review]
In general all fine, i have only question about sqlalchemy.
Also, wen need suggestion from @ilia-tsyplenkov : what we can do with `ml_handler_wrapper.py`?

**Comment by ZoranPandovski** [comment]
Closing this for now. @ShrutiC-git you put a lot of effort into this so I will send you an email and discuss this

**Pull request closed without merging**